src/openforms/validations/api/serializers.py

```diff
@@ -33,3 +37,7 @@ class ValidationPluginSerializer(serializers.Serializer):
         label=_("Label"),
         help_text=_("The human-readable name for a plugin."),
     )
+    components = serializers.ListField(
+        label=_("Components"),
+        help_text=_("The components for which the plugin is relevant."),
+    )
```

Comment:
I think that `serializers.CharField(many=True)` is more appropriate if that's possible, otherwise you want to specify the `child` argument of the `ListField` to properly type the items. That should fix the OpenAPI schema issues.